main.py

```diff
@@ -109,54 +58,115 @@ def main(cfg: DictConfig):
         cfg.finetune.test_batch_size,
     )
 
+    logger.info("Calculating metrics on base model")
+    # TODO cache the base model metrics, and only recompute if the cache does not exist
     if not cfg.pretrained:
-        train(
-            model,
+        base_pre_loss = train(
+            base_model,
             device,
             pre_train_loader,
             cfg.pretrain.epochs,
             learning_rate=cfg.pretrain.lr,
         )
 
-    _, pre_acc = test(model, device, pre_test_loader)
+    # prepare the model for the unadapt method
+    ufm_model = copy.deepcopy(base_model)
 
-    # Finetune model
-    fine_model = copy.deepcopy(model)
-    model_fine_losses = train(
-        fine_model,
+    if cfg.metric == "accuracy":
+        _, base_pre_acc = metrics.calculate_loss_and_accuracy(
+            base_model, device, pre_test_loader
+        )
+    elif cfg.metric == "llm_tasks":
+        metrics.run_llm_benchmark(base_model, logger=logger, tag="base_model")
+
+    base_fine_losses = train(
+        base_model,
         device,
         fine_train_loader,
         cfg.finetune.epochs,
         learning_rate=cfg.finetune.lr,
     )
-    _, model_fine_acc = test(fine_model, device, fine_test_loader)
+    wandb.log({"base_fine_losses": base_fine_losses})
 
-    for unadapt_config in cfg.unadapt:
-        ufm_pre_acc, ufm_fine_acc, ufm_fine_losses = calculate_unadaptability_metrics(
-            model,
-            unadapt_config,
+    if cfg.metric == "accuracy":
+        _, base_fine_acc = metrics.calculate_loss_and_accuracy(
+            base_model,
             device,
-            config,
-            pre_train_loader,
-            pre_test_loader,
-            fine_train_loader,
             fine_test_loader,
         )
-        pre_acc_ratio = ufm_pre_acc / pre_acc
-        fine_acc_ratio = ufm_fine_acc / model_fine_acc
-        loss_gap_ratio = calculate_loss_gap_ratio(ufm_fine_losses, model_fine_losses)
+    elif cfg.metric == "llm_tasks":
+        metrics.run_llm_benchmark(base_model, logger=logger, tag="base_model_finetuned")
+
+    # clean base model since we don't need it anymore
+    del base_model
```

Comment:
We probably want to keep the base model for the unadapt method so we can do things like a KL divergence penalty.

Reply:
`ufm_model = copy.deepcopy(base_model)`
I am copying the model in line 73, and using that model to test unadapt methods.
Currently main.py finetunes in both the base model and unadapt model in one execution.
I think it would be better to instead just do one as you mentioned in the doc.

> The first time we run our script without an unadaptability method specified, it should benchmark and save the results to disk from benchmarking and fine-tuning the base model for comparison.

Reply:
Delete both this `del` line and the deepcopy. We should pass the base model into the unadapt method and let the individual unadapt method decide if it wants to copy the base mode before returning just the unadapted model (as is shown in the latest main.py).